### PR TITLE
Added ability to set container ulimit.

### PR DESCRIPTION
**Pull request by zsuzhengdu**
PR for https://github.com/signalfx/maestro-ng/issues/161

### CODE REVIEW

**Comment by iangkent** [comment]
+1

File: maestro/entities.py [hunk 1]
```diff
@@ -851,6 +855,23 @@ def _parse_lifecycle(self, lifecycles):
...
+        else:
+            result = []
+            for name, value in ulimits.items():
+                ulimit = {}
+                if isinstance(value, dict):
+                    ulimit.update({'name': name})
+                    ulimit.update(value)
+                elif isinstance(value, int):
+                    ulimit.update({'name': name})
+                    ulimit.update({'hard': value, 'soft': value})
```
Comment (mpetazzoni) [inline maestro/entities.py]: You should have a `else: continue` in here so you don't append an empty `ulimit` dictionary to the `result` list

File: maestro/entities.py [hunk 2]
```diff
@@ -851,6 +855,23 @@ def _parse_lifecycle(self, lifecycles):
...
+                result.append(ulimit)
```
Comment (mpetazzoni) [inline maestro/entities.py]: Hmm, I'd rewrite this whole thing like this:

```python
result = []
for name, value in ulimits.items():
    ulimit = {'name': name}
    if isinstance(value, dict):
        ulimit.update(value)
    elif isinstance(value, int):
        ulimit.update({'hard': value, 'soft': value})
    else:
        continue
    result.append(ulimit)
return result
```

File: maestro/entities.py [hunk 3]
```diff
@@ -851,6 +855,23 @@ def _parse_lifecycle(self, lifecycles):
 
         return checkers
 
+    def _parse_ulimits(self, ulimits):
+        """Parse ulimits"""
+        if ulimits is None:
+            return None
+        else:
```
Comment (mpetazzoni) [inline maestro/entities.py]: You don't need the else here, so you can save an indentation level

**Comment by mpetazzoni** [comment]
Thanks for the contribution! Can you also provide a unit test for this?

**Comment by mpetazzoni** [comment]
Thanks. Do we care about validating the name of the limits? Is there a known set of valid limit names?

**Comment by zsuzhengdu** [comment]
We do not need worry about validating the name of limits, 'docker' engine will do.  If an invalid type is provided, following error will be threw out with input 

```
ulimits:
  kobe: 65535
``` 

```
  File "/usr/lib/python2.7/site-packages/docker/api/container.py", line 368, in start
    self._raise_for_status(res)
  File "/usr/lib/python2.7/site-packages/docker/client.py", line 174, in _raise_for_status
    raise errors.APIError(e, response, explanation=explanation)
APIError: 500 Server Error: Internal Server Error ("{"message":"oci runtime error: Wrong rlimit value: RLIMIT_KOBE"}")
```

Here is a list of 'supported' names. 
https://github.com/docker/docker/blob/1c32bed48c160342871d4f1470dd4d62b4baa9a5/vendor/src/github.com/docker/go-units/ulimit.go#L23

**Comment by mpetazzoni** [comment]
Thanks!

**Comment by zsuzhengdu** [comment]
Thank you for your insightful code sample and  comments!

**Pull request merged**